packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+		// min and max bounderies and update the previous state reference.
+		values.forEach(valueType => {
+			if (this.isPropertyUpdated(valueType)) {
+				const normalizedValue = SliderBase.clipValue(this[valueType], this.min, this.max);
+				this.updateValue(valueType, normalizedValue);
+				this.storePropertyState(valueType);
+			}
+		});
+	}
+
+	/**
+	 * In order to always keep the visual UI representation and the internal
+	 * state in sync, the component has a 'state storage' that is updated when the
+	 * current state is changed due to a user action.
+	 *
+	 * Check if the previously saved state is outdated. That would mean
+	 * a property has been changed programatically because the previous state
+	 * is always updated in the interaction handlers.
+	 *
+	 * Will return true if any of the properties is not equal to its previously
+	 * stored value.
+	 *
+	 * @protected
+	 */
+	isCurrentStateOutdated() {
+		return Object.entries(this._stateStorage).some(([propName, propValue]) => this[propName] !== propValue);
+	}
+
+	/**
+	 * Returns the last stored value of a property
+	 *
+	 * @protected
+	 */
+	getStoredPropertyState(property) {
+		return this._stateStorage[property];
+	}
+
+	/**
+	 * Check if one or more properties have been updated compared to their last
+	 * saved values in the state storage.
+	 *
+	 * @protected
+	 */
+	isPropertyUpdated(...properties) {
+		return properties.some(prop => this.getStoredPropertyState(prop) !== this[prop]);
+	}
+
+	/**
+	 * Updates the previously saved in the _stateStorage values of one or more properties.
+	 *
+	 * @protected
+	 */
+	storePropertyState(...props) {
+		props.forEach(property => {
+			this._stateStorage[property] = this[property];
+		});
+	}
+
+	/**
+	 * Returns the start side of a direction - left for LTR, right for RTL
+	 *
+	 * @protected
+	 */
+	get directionStart() {
+		return this.effectiveDir === "rtl" ? "right" : "left";
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.tickmarks || !this.step) {
+			return;
+		}
+
+		// Convert number values to strings to let the CSS do calculations better
+		// rounding/subpixel behavior" and the most precise tickmarks distribution
+		const maxStr = String(max);
+		const minStr = String(min);
+		const stepStr = String(step);
+		const tickmarkWidth = "1px";
+
+		// There is a CSS bug with the 'currentcolor' value of a CSS gradient that does not
+		// respect the variable for more than one theme. It has to be set here for now.
+		const currentTheme = getTheme();
+		const currentColor = this.TICKMARK_COLOR_MAP[currentTheme];
+
+		this._tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
+		this._hiddenTickmarks = false;
+
+		// Transparent CSS gradient background
+		const tickmarksGradientBase = `linear-gradient(to right, ${currentColor} ${tickmarkWidth}, transparent 0) `;
```

Comment:
I am sure it is done this way for a reason, but it's worth discussing it briefly as it is not the straightforward.

Reply:
We draw the tick marks as gradient style only with CSS letting the CSS calculate the positioning and the distribution of the marks. That way we avoid creating an HTML element for every tick mark and calculate its position in JS and apply it to the element afterwards (like in classic UI5 sliders). I think it is a cleaner and faster solution.